### DISCUSSION TITLE
Web Server Not Working Correctly

### DISCUSSION THREAD

**Discussion by kglasnapp** [PlummersSoftwareLLC/NightDriverStrip#17] [Q&A]
Thanks so much Dave this is a great tool. I have been able to get the basic demo to work and I can connect to the web server but the html that is sent to my browser is from an old web application that I used on the ESP 32. How do I update the filesystem on the ESP32 to use the web server you provided?

**Comment by MythicalForce** [comment]
Format spiffs and upload the content in /data folder

**Reply by kglasnapp** [reply]
Thanks -- I was able to do that task in Visual Studio Code -- by clicking on the Platform IO Icon on the left side, select project tasks, select demo, select platform, select upload file system image. The /data folder had the code for the web page. If did not have a file system on my EPS32, I might have needed to select "Build File System Image"

**Reply by jurjanbrust** [reply]
I think that it would be great to add this to the readme.md

**Reply by BeldrothTheGold** [reply]
I submitted a PR to do just that. https://github.com/PlummersSoftwareLLC/NightDriverStrip/pull/65

**Comment by BeldrothTheGold** [comment]
The below section was just merged into the main project README.md. Keep in mind that the current partitions_custom.csv in main has some math errors, so this will not work unless you have 8M of flash, until PR #64 is merged. You can use the version of partitions_custom.csv from that PR if you like, and you should be good. Let me know if there are any issues. 

## Webserver Setup
To use the built-in webserver, you will need to build and upload the SPIFFS image to your board's flash using platformio.
You can do this using the platformio user interface, or using the pio command line tool
```
pio run --target buildfs --environment <project name>
pio run --target uploadfs --environment <project name>
```
